Paintroid/src/test/java/org/catrobat/paintroid/test/presenter/MainActivityPresenterTest.java

```diff
@@ -103,6 +114,7 @@
 	@Mock
 	private MainActivityContracts.BottomBarViewHolder bottomBarViewHolder;
 
+	@Spy
```

Comment:
Presenter should be a regular instance, not a spy. Is this somehow needed?

Reply:
Yes. 
It's needed for the 4 tests which verify that the previous action is repeated after the permission was granted. 1035-1103 
I can try to find another way to test that if needed.

Reply:
:980
:1011
:1042
:1075

After splitting.